src/shogun/mathematics/linalg/internal/implementation/MeanEigen3.h

```diff
@@ -93,10 +93,19 @@ struct mean
 	typedef typename int2float<T>::floatType ReturnType;
 
 	/**
-	 * Method that computes the vector/matrix mean
+	 * Method that computes the vector mean
 	 *
-	 * @param a vector/matrix whose mean has to be computed
-	 * @return the vector/matrix mean \f$\mean_i a_i\f$
+	 * @param a vector whose mean has to be computed
+	 * @return the vector mean \f$\mean_i a_i\f$
+	 */
+	static ReturnType compute(Matrix a);
+
+	/**
+	 * Method that computes the matrix mean
+	 *
+	 * @param a matrix whose mean has to be computed
+	 * @param no_diag if true, diagonal entries are excluded from the mean (default - false) 
+	 * @return the matrix mean \f$\mu_{i,j}m_{i,j}\f$
```

Comment:
what in particular?
This should be 1/N^2 \sum_{i,j=1}^N m_{i,j} (if mean is included)